docs/csharp/language-reference/compiler-messages/cs8410.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Compiler Error CS8410
+ms.date: 07/11/2020
+f1_keywords:
+  - "CS8410"
+helpviewer_keywords:
+  - "CS8410"
+author: Youssef1313
+---
+# Compiler Error CS8410
+
+'type': type used in an asynchronous using statement must be implicitly convertible to 'System.IAsyncDisposable' or implement a suitable 'DisposeAsync' method.
+
+The expression inside an `await using` statement must have a `DisposeAsync` method.
+
+## To correct this error
+
+Use `using` instead of `await using`.
```

Comment:
This is not the correct messaging for this error message. This actually corresponds to `ERR_NoConvToIAsyncDispWrongAsync = 8417`, see [resource](https://github.com/dotnet/roslyn/blob/dbdae0b756975b86f3411c3ec9e01df4afbe1fc8/src/Compilers/CSharp/Portable/CSharpResources.resx#L3004-L3009)

Reply:
@IEvangelist, this line isn't referred to as being the message of the error. It just suggests the common fix for this error.

Reply:
[See this please.](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKBQBgAIUMBWAbl1xIDZi1iMB2Q3Ab10K8bpQA5idALIQAlgDsAFAEpWOboQ7yF3FAE5iRAG4QEhBHAgATRIQC8hcXADuhAEqGTCGZWUriGgoR16joqAAOAPZQEMAANnAOxqYWVrYAIv7BoRFRjogunO5yOZreuoQAHuaWNvYZztKueV4+hACepfGESYEhYZHRTlluhAC+uIM41AzdpmwDVHgMbSmd6TF6IIQAknMdaezZXCgAzMQALK3JIXAyilM4/UA=)

Reply:
Oh got it. Then what you recommend to put under "To correct this error"?

Reply:
The only thing that makes sense is to remove both the `await` and the `using`. If the type is neither an implementation of `IDisposable` or `IAsyncDisposable`, I think that's the only logic suggestion that can be made. @BillWagner, do you agree?

Reply:
Yes, I agree. If a type is not disposable (synchronously or asynchronously), the only fix is to remove the `using`

Reply:
@BillWagner @IEvangelist

I think we may also add "implement a suitable `DisposeAsync` method" as another option. What do you think?

Reply:
Here is an updated version of the [SharpLab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKBQBgAIUMBWAbl1xIDZi1iMB2Q3Ab10K8bpQA5idALIQAlgDsAFAEpWOboQ7yF3APSrCAYQDKfACwYiAWkIwAFqKiFLhAHIB5ACqEAZqIAeEYABs4hAO6i5qZmfghwYAD2AG4Q3oSRLoQABigAnMmcKlzpxESxCIThEAAmiIQAvITicP6EAEpwpYgylMrZWdnqdpGEiAiRCJ0qBUVNZQgMVTV1jc0IrcMKS2oaOvrMhCbmNjZunj5+gcHmYRExcQlJqRkrOWl5hKMllgAOkVBevnMTldW1hAAIm8Pl84D8WtI2tkuHdCN1bL1+oM4QQnhBCi8oO9PocIZM/jMgSDcd9xpDodwAL64Gk4agMfGKQh0hnE7GgvHkwogQgASWBHNJcHYwxQAGZiHp2Ti4DJmXSqUA===)

Reply:
These are very useful when documenting cs8417. We may even link the two pages as they're a bit related.